### ISSUE TITLE
[Question] what difference between `braindecode.EEGClassifier` and `skorch.NeuralNetClassifier`

### ISSUE THREAD

**Issue by edw4rdyao** [braindecode/braindecode#560]
hi all and @bruAristimunha , I use braindecode to train my model and dataset (**braindecode is a great lib for eeg**), I found that the valid accuracy is different when I use `braindecode.EEGClassifier` and `skorch.NeuralNetClassifier`( `braindecode.EEGClassifier`  valid accuracy higher 10% than skorch.NeuralNetClassifier, I have run for many times), 

So I want to ask the following questions:
-  If I don't use `crop training`, will the training process and results of the two classifiers(`braindecode.EEGClassifier` and `skorch.NeuralNetClassifier`) be the same? (**when the data, split, and training hyperparameter configurations are the same**)
- Is there anything I misunderstand(or wrong usage in code below) about EEGClassifier?

Detail is below:

The code for `braindecode.EEGClassifier`:

```python
...
n_channels = 6
n_times = 4000
n_classes = 3
n_epochs = 300
lr = 0.0005
batch_size = 64

my_model = MyModel(n_chans=n_channels, n_outputs=n_classes, n_times=n_times)

clf = EEGClassifier(
    module=my_model,
    criterion=torch.nn.CrossEntropyLoss,
    optimizer=torch.optim.Adam,
    optimizer__lr= lr,
    train_split=ValidSplit(0.2, stratified=True, random_state=1), # make sure the valid set is same every run
    batch_size= batch_size,
    callbacks=[],
    device='cuda' if cuda else 'cpu',
    verbose=1,
    max_epochs=n_epochs
)
# dataset is EEGWindowsDataset
dataset_X = SliceDataset(dataset, idx=0)
dataset_y = SliceDataset(dataset, idx=1)
clf.fit(X=dataset_X, y=dataset_y)
...
```
Result:
```
 epoch    train_loss    valid_acc    valid_loss     dur
-------  ------------  -----------  ------------  ------
      1        1.2402       0.3778        1.0950  0.5500
      2        1.1308       0.3444        1.0893  0.0914
      3        1.0405       0.4444        1.0883  0.0920
      4        0.9974       0.5111        1.0692  0.0827
      5        0.9654       0.4444        1.0519  0.0884
...
    280        0.0975       0.8222        0.7976  0.0839
    281        0.1059       0.8111        0.7476  0.0823
    282        0.0811       0.8556        0.7732  0.0806
    283        0.0719       0.8333        0.8144  0.0816
    284        0.1122       0.8222        0.7814  0.0805
    285        0.0989       0.8333        0.7546  0.0801
    286        0.1051       0.8222        0.8042  0.0837
    287        0.0812       0.8000        0.8845  0.0826
```

The code for `skorch.NeuralNetClassifier`(**In fact and sure that I only changed the type of classifier**):

```python
...
n_channels = 6
n_times = 4000
n_classes = 3
n_epochs = 300
lr = 0.0005
batch_size = 64

my_model = MyModel(n_chans=n_channels, n_outputs=n_classes, n_times=n_times)

clf = NeuralNetClassifier(
    module=my_model,
    criterion=torch.nn.CrossEntropyLoss,
    optimizer=torch.optim.Adam,
    optimizer__lr= lr,
    train_split=ValidSplit(0.2, stratified=True, random_state=1), # make sure the valid set is same every run
    batch_size= batch_size,
    callbacks=[],
    device='cuda' if cuda else 'cpu',
    verbose=1,
    max_epochs=n_epochs
)
# dataset is EEGWindowsDataset
dataset_X = SliceDataset(dataset, idx=0)
dataset_y = SliceDataset(dataset, idx=1)
clf.fit(X=dataset_X, y=dataset_y)
...
```
Result:
```
epoch    train_loss    valid_acc    valid_loss     dur
-------  ------------  -----------  ------------  ------
      1        1.2153       0.3889        1.0980  0.3488
      2        1.1214       0.3444        1.0980  0.1762
      3        1.0290       0.3444        1.1026  0.1838
      4        0.9745       0.3444        1.1161  0.1270
      5        0.9163       0.3444        1.1119  0.1001
...
    280        0.1844       0.6333        1.9210  0.0964
    281        0.1822       0.6667        1.1688  0.0932
    282        0.1139       0.7111        1.2593  0.0983
    283        0.1336       0.6556        1.2219  0.0929
    284        0.0960       0.6444        1.3905  0.0929
    285        0.0772       0.6667        1.0540  0.1007
    286        0.0783       0.7111        1.1803  0.1000
    287        0.0784       0.6889        1.0688  0.0968
    288        0.0474       0.6111        2.3137  0.0946
    289        0.0500       0.6778        1.9161  0.0948
    290        0.0387       0.7000        1.6264  0.0967
```

**Comment by edw4rdyao** [comment]
Thank you @bruAristimunha  I found the reason. The main problem is that the default training dataset of `braindecode.EEGClassifier` is shuffled, but the `skorch.NeuralNetClassifier`  is not(default `iterator_train__shuffle=False`).

https://github.com/braindecode/braindecode/blob/22da938bec9445aaa7a7a58a6abcfb14bb2d1d71/braindecode/classifier.py#L54-L56

**Comment by robintibor** [comment]
ah right great you found it was checking but didn't realize that's the difference, good to keep in mind
